File: Ship.js
var file = require('./ships.json');

class Ship {
  constructor(name, weapon, aura, zen) {
    this.name = String(name);
    this.weapon = String(weapon);
    this.aura = String(aura);
    this.zen = String(zen);
  };

  tostring(){
    return ("**" + this.name+":** Weapon: *" + this.weapon +"*, Aura: *" + this.aura + "*, Zen: *" + this.zen + "*");
  }
}

function getallships() {
  var list = "List of all ship:\n";
  for (var ship in allshipsf.ships) {
    if(String(ship.name) != ''){
      list += "\n" + ("**" + String(ship.name)+":** Weapon: *" + String(ship.weapon) +"*, Aura: *" + String(ship.aura) + "*, Zen: *" + String(ship.zen) + "*");
    }
  }
}

class AllShips{
  this.ships = {};
  static SHIPS() {return ships};
  constructor(){
    for (var ship in allshipsf.ships) {
      if(String(ship.name) != ''){
        this.ships.push(new Ship(ship.name, ship.weapon, ship.aura, ship.zen));
      }      
    }
  };   
 }

module.exports = {
  Ship: Ship,
  getallships: getallships
}

File: src/main/java/jn/rocbot/ships/Ship.java
package jn.rocbot.ships;

import jn.rocbot.commands.Formatter;
import jn.rocbot.info.ShipNotFoundException;
import jn.rocbot.info.ShipStore;

public class Ship implements Formatter{
    public final String name;
    public final String weapon;
    public final String aura;
    public final String zen;
    public final RARITY rarity;

    public Ship(String name, String weapon, String aura, String zen, RARITY rarity) {
        this.name = name;
        this.weapon = weapon;
        this.aura = aura;
        this.zen = zen;
        this.rarity = rarity;
    }

    public String simpleToString(){
        return rarity.toEmoji() + " " + bold(name + ":") + " Weapon: " + italic(weapon) + ", Aura: " + italic(aura) + ", Zen: " + italic(zen);
    }

    public static boolean isShip(String name){
        try {
            return ShipStore.getShip(name) != null;
        } catch (ShipNotFoundException e) {
            return false;
        }
    }
}